### ISSUE TITLE
Constructors for auto generated classes?

### ISSUE THREAD

**Issue by Tbilgere-td** [profusion/sgqlc#42]
We're using sgqlc to interact with a GQL service we've built and love the removal of dict-of-stuff navigation in the response objects and field selections.  We can't quite figure out how to do the same for complex argument/input values though.
As an example:
```python
class FilterDateInput(sgqlc.types.Input):
    start = sgqlc.types.Field(String, graphql_name='start')
    end = sgqlc.types.Field(String, graphql_name='end')
class FilterValueInput(sgqlc.types.Input):
    value = sgqlc.types.Field(String, graphql_name='value')
class FilterInput(sgqlc.types.Input):
    date = sgqlc.types.Field(FilterDateInput, graphql_name='date')
    id = sgqlc.types.Field(FilterInput, graphql_name='id')
```
Ideally we would be able to generate a filter by:
```python
gql_filter = FilterInput(
    date=FilterDateInput(start='04-12-2012', end='04-15-2012'), 
    id=FilterValueInput(value='A17')
)
operation.queryThings(filter=gql_filter, user="bill")
```
This would benefit from tool tips and autocomplete where as the current method we're aware of is:
```python
gql_filter = { 
    'date': {'start':'04-12-2012', 'end':'04-12-2012'}, 
    'id':{'value':'A17'}
}
operation.queryThings(filter=gql_filter, user="bill")
```
Which as the schema grows become increasingly unwieldy. Is there a way that we are missing to do this or is there a way to generate classes that will act in this way using introspection of the sgqlc’s autogenerated classes?

**Comment by barbieri** [comment]
I focused on the read-side of the data, it's basically implemented by the `sgqlc.operation` and selectors, so I think the creation side will need some specific work to allow that.

Shouldn't be hard as `input`  types are much simpler and particularly if you stick with `__init__` and `__setattr__` to set values.

I did this for some other project (customer private), not hard... but takes some time I don't  have at this moment. This month is being particularly busy at work. I'll try to get back to sqglc development in June.
